File: Introductory Problems/Bit Strings/code.py
n = int(input())
M = 1_00_00_00_000 + 7

def bin_pow(a,b):
    ans = 1
    while b > 0:
        if b & 1:
            ans = (ans * a) % M
        a = (a * a) % M
        b = b >> 1
    return ans

print(bin_pow(2,n))

File: Introductory Problems/Bit Strings/code.js
const fs = require('fs')
const input = fs.readFileSync(0, 'utf-8').trim().split('\n');
const _n = parseInt(input[0], 10)
const n = BigInt(_n)


function bin_pow(a, b, MOD = 1_00_00_00_007n) {
    let ans = 1n;
    while (b > 0n) {
        if (b&1n) {
            ans = (ans * a)%MOD;
        }
        a = (a*a) % MOD;
        b = b >> 1n;
    }
    return ans;
}

console.log(bin_pow(2n, n).toString());